### PR TITLE
Azure AD -> Microsoft Entra ID

### LINKED ISSUE #13990: Rename Azure AD wording to Microsoft Entra ID
### Is your feature request related to a problem? Please describe.
Since naming products was never a strong point of Microsoft, they're renaming Azure AD to Microsoft Entra ID, see https://techcommunity.microsoft.com/t5/microsoft-entra-azure-ad-blog/azure-ad-is-becoming-microsoft-entra-id/ba-p/2520436. This necessitates that we also rename the Azure AD-related terminology.

### Describe the solution you'd like
Rename the user-facing strings (but not technical IDs) of the [Azure AD feature](https://docs.orchardcore.net/en/latest/docs/reference/modules/Microsoft.Authentication/#azure-active-directory), both in the code and in the docs. Also include the old name though, so people searching for Azure AD will still be able to find everything. Mike did something similar under [this PR](https://github.com/OrchardCMS/OrchardCore/pull/13852) for Facebook -> Meta.

### Describe alternatives you've considered
Also renaming the technical IDs, but that would be a breaking change. It'd get rid of some technical debt though.

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #13990

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -106,7 +106,7 @@ nav:
       - Understanding Content Definitions Stores: docs/guides/content-definitions/README.md
       - Change password configuration: docs/guides/password-configuration/README.md
       - Change lockout configuration: docs/guides/lockout-configuration/README.md
-      - Azure AD integration: docs/guides/azuread-integration/README.md
+      - Microsoft Entra ID / Azure AD integration: docs/guides/microsoft-entra-id-integration/README.md
```
Comment (Piedone) [inline mkdocs.yml:109]:
```suggestion
      - Microsoft Entra ID (Azure AD) integration: docs/guides/microsoft-entra-id-integration/README.md
```

File: src/docs/guides/microsoft-entra-id-integration/README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Use Azure AD as an external identity provider
+# Use Microsoft Entra ID / Azure AD as an external identity provider
```
Comment (Piedone) [inline src/docs/guides/microsoft-entra-id-integration/README.md:1]: Run the docs locally, see https://docs.orchardcore.net/en/latest/docs/topics/docs-contributions/, to see if anything got broken. There's this one at least:

image

**Review by Piedone — CHANGES_REQUESTED** [review]
Please don't resolve my comments when you're done with something so I can track feedback (except for code suggestions that get resolve automatically). However, you can keep track whether you've addressed something with a reaction under the message for example if you want to.

**Review by MikeAlhayek — APPROVED** [review]

**Comment by Piedone** [comment]
Why did you merge this, @agriffard? I didn't approve it, and there are pending comments of mine that Hisham couldn't address.

**Comment by hishamco** [comment]
Thanks, @MikeAlhayek for the updates that you did

> Why did you merge this, @agriffard? I didn't approve it, and there are pending comments of mine that Hisham couldn't address.

Seems he merge when Mike approved :) anyhow I will check your comments Zoltan

**Comment by agriffard** [comment]
> Why did you merge this, @agriffard? I didn't approve it, and there are pending comments of mine that Hisham couldn't address.

Sorry @Piedone , I saw that @MikeAlhayek had made some changes related to your comments and approved the PR so I thought it was ready to merge.

**Comment by Piedone** [comment]
I see, no problem.

**Pull request merged**